include/nbl/builtin/glsl/ext/FFT/default_compute_fft.comp

```diff
@@ -0,0 +1,127 @@
+#ifndef _NBL_GLSL_EXT_DEFAULT_COMPUTE_FFT_INCLUDED_
+#define _NBL_GLSL_EXT_DEFAULT_COMPUTE_FFT_INCLUDED_
+
+// WorkGroup Size
+
+#ifndef USE_SSBO_FOR_INPUT
+#error "USE_SSBO_FOR_INPUT should be defined."
+#endif
+
+#ifndef _NBL_GLSL_EXT_FFT_BLOCK_SIZE_X_DEFINED_
+#define _NBL_GLSL_EXT_FFT_BLOCK_SIZE_X_DEFINED_ 256
+#endif
+
+#ifndef _NBL_GLSL_EXT_FFT_BLOCK_SIZE_Y_DEFINED_
+#define _NBL_GLSL_EXT_FFT_BLOCK_SIZE_Y_DEFINED_ 1
+#endif
+
+#ifndef _NBL_GLSL_EXT_FFT_BLOCK_SIZE_Z_DEFINED_
+#define _NBL_GLSL_EXT_FFT_BLOCK_SIZE_Z_DEFINED_ 1
+#endif
+
+#define _NBL_GLSL_WORKGROUP_SIZE_ (_NBL_GLSL_EXT_FFT_BLOCK_SIZE_X_DEFINED_*_NBL_GLSL_EXT_FFT_BLOCK_SIZE_Y_DEFINED_*_NBL_GLSL_EXT_FFT_BLOCK_SIZE_Z_DEFINED_)
+
+layout(local_size_x=_NBL_GLSL_EXT_FFT_BLOCK_SIZE_X_DEFINED_, local_size_y=_NBL_GLSL_EXT_FFT_BLOCK_SIZE_Y_DEFINED_, local_size_z=_NBL_GLSL_EXT_FFT_BLOCK_SIZE_Z_DEFINED_) in;
+
+ 
+#define _NBL_GLSL_EXT_FFT_GET_DATA_DEFINED_
+#define _NBL_GLSL_EXT_FFT_SET_DATA_DEFINED_
+#define _NBL_GLSL_EXT_FFT_GET_PADDED_DATA_DEFINED_
+#include "nbl/builtin/glsl/ext/FFT/fft.glsl"
+
+// Input Descriptor
+
+struct nbl_glsl_ext_FFT_input_t
+{
+	vec2 complex_value;
+};
+
+#ifndef _NBL_GLSL_EXT_FFT_INPUT_SET_DEFINED_
+#define _NBL_GLSL_EXT_FFT_INPUT_SET_DEFINED_ 0
+#endif
+
+#ifndef _NBL_GLSL_EXT_FFT_INPUT_BINDING_DEFINED_
+#define _NBL_GLSL_EXT_FFT_INPUT_BINDING_DEFINED_ 0
+#endif
+
+#ifndef _NBL_GLSL_EXT_FFT_INPUT_DESCRIPTOR_DEFINED_
+
+#if USE_SSBO_FOR_INPUT > 0
+#define _NBL_GLSL_EXT_FFT_INPUT_DESCRIPTOR_DEFINED_
+layout(set=_NBL_GLSL_EXT_FFT_INPUT_SET_DEFINED_, binding=_NBL_GLSL_EXT_FFT_INPUT_BINDING_DEFINED_) readonly restrict buffer InputBuffer
+{
+	nbl_glsl_ext_FFT_input_t inData[];
+};
+#else 
+layout(set=_NBL_GLSL_EXT_FFT_INPUT_SET_DEFINED_, binding=_NBL_GLSL_EXT_FFT_INPUT_BINDING_DEFINED_) uniform sampler2D inputImage;
+#endif
+
+#endif
+
+// Output Descriptor
+
+struct nbl_glsl_ext_FFT_output_t
+{
+	vec2 complex_value;
+};
+
+
+#ifndef _NBL_GLSL_EXT_FFT_OUTPUT_SET_DEFINED_
+#define _NBL_GLSL_EXT_FFT_OUTPUT_SET_DEFINED_ 0
+#endif
+
+#ifndef _NBL_GLSL_EXT_FFT_OUTPUT_BINDING_DEFINED_
+#define _NBL_GLSL_EXT_FFT_OUTPUT_BINDING_DEFINED_ 1
+#endif
+
+#ifndef _NBL_GLSL_EXT_FFT_OUTPUT_DESCRIPTOR_DEFINED_
+#define _NBL_GLSL_EXT_FFT_OUTPUT_DESCRIPTOR_DEFINED_
+layout(set=_NBL_GLSL_EXT_FFT_OUTPUT_SET_DEFINED_, binding=_NBL_GLSL_EXT_FFT_OUTPUT_BINDING_DEFINED_) restrict buffer OutputBuffer
+{
+	nbl_glsl_ext_FFT_output_t outData[];
+};
+#endif
+
+// Get/Set Data Function
+
+nbl_glsl_complex nbl_glsl_ext_FFT_getData(in uvec3 coordinate, in uint channel)
+{
+	vec2 retValue = vec2(0, 0);
+#if USE_SSBO_FOR_INPUT > 0
+	uvec3 dimension = pc.dimension;
+	uint index = channel * (dimension.x * dimension.y * dimension.z) + coordinate.z * (dimension.x * dimension.y) + coordinate.y * (dimension.x) + coordinate.x;
```

Comment:
ok here you better repeat or clamp the `coordinate` according to the paratemeters, cause this aint a sampler

Reply:
ah ok I see the clamping happens in `nbl_glsl_ext_FFT_getPaddedData` but you could skip it for textures, and just use `textureLod` + the correct sampler and border color
https://github.com/Devsh-Graphics-Programming/Nabla/pull/63#discussion_r572230516